File: src/lib/pjecalc/core/dominio/calculo/relatorio/seguro-desemprego-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — SeguroDesempregoJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.SeguroDesempregoJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/SeguroDesempregoJRAdapter.java
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class SeguroDesempregoJRAdapter extends JRAdapter {
  abstract getParcelas(): JRAdapterDataSource<SeguroDesempregoParcelaAdapter>;
  abstract getBaseMediaUltimosSalarios(): Decimal;
  abstract getQuantidadeParcelas(): number;
  abstract getValorTotal(): Decimal;
}

export abstract class SeguroDesempregoParcelaAdapter extends JRAdapter {
  abstract getNumeroParcela(): number;
  abstract getCompetencia(): Date;
  abstract getValor(): Decimal;
  abstract getIndiceCorrecao(): Decimal;
  abstract getValorCorrigido(): Decimal;
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/SeguroDesempregoJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.negocio.constantes.TipoValorSeguroDesempregoEnum;
import java.math.BigDecimal;
import java.util.Date;

public abstract class SeguroDesempregoJRAdapter
extends JRAdapter {
    public abstract Date getOcorrencia();

    public abstract String getFormula();

    public abstract BigDecimal getSalarioReferencia();

    public abstract BigDecimal getValorParcela();

    public abstract Integer getQuantidadeParcelas();

    public abstract BigDecimal getDevido();

    public abstract BigDecimal getIndice();

    public abstract BigDecimal getValorCorrigido();

    public abstract BigDecimal getJuros();

    public abstract BigDecimal getTotal();

    public abstract TipoValorSeguroDesempregoEnum getTipoValorDoSeguroDesemprego();
}

